snuba/datasets/discover.py

```diff
@@ -263,47 +192,41 @@ def __init__(self) -> None:
             ]
         )
 
+        # TODO: Move the storage definition out of events and transactions dataset so
+        # we can reuse the class itself and avoid the isinstance call.
+        events_dataset = get_dataset(EVENTS)
+        assert isinstance(events_dataset, EventsDataset)
+        transactions_dataset = get_dataset(TRANSACTIONS)
+        assert isinstance(transactions_dataset, TransactionsDataset)
+        storage_selector = DiscoverQueryStorageSelector(
+            events_table=events_dataset.get_storage(),
+            transactions_table=transactions_dataset.get_storage(),
```

Comment:
It's kinda confusing that dataset has both `get_all_storages()` and sometimes `get_storage()` methods

Reply:
get_storage is just an intermediate solution to simplify this PR. IT disappears in #829

Reply:
This would have been much clearer if it was noted near the definition of `get_storage`, this tripped me up on a second reading of this change as well.

Reply:
ok, will add a comment there.